conda_smithy/configure_feedstock.py

```diff
@@ -878,6 +881,39 @@ def migrate_combined_spec(combined_spec, forge_dir, config, forge_config):
     return combined_spec
 
 
+def reduce_variants(recipe_path, config, input_variants):
+    """Subset of render_recipe to compute reduced variant matrix
+
+    large numbers of unused variants greatly increase render time
+    """
+
+    # from render_recipe
+    with conda_build.render.open_recipe(recipe_path) as recipe:
+        metadata = MetaData(str(recipe), config=config)
+
+    # from distribute_variants
+    # explode variants dict to list of variants
+    variants = initial_variants = conda_build.variants.get_package_variants(
+        metadata, variants=input_variants
+    )
+    logger.debug(f"Starting with {len(initial_variants)} variants")
+    metadata.config.variant = variants[0]
+    metadata.config.variants = variants
+    all_used = metadata.get_used_vars(force_global=True)
+    # trim unused dimensions, these cost a lot in render_recipe!
+    all_used.update(ALWAYS_KEEP_KEYS)
+    all_used.update(
+        {"target_platform", "extend_keys", "ignore_build_only_deps"}
+    )
+    new_input_variants = input_variants.copy()
+    for key, value in input_variants.items():
+        if len(value) > 1 and key not in all_used:
```

Comment:
Is there a reason to not remove keys with dimension 1 here too if they are unused?

Reply:
No, if this is rigorous and always guarantees to get all used keys every time. Since increased dimensionality is the real source of cost, only considering keys which increase dimensionality reduces the chances of imprecise logic having an impact (see comment below for a little more detail)

Reply:
hundreds of other unused keys contributes to the cost as well, so removing all unused keys here would be ideal and have a benefit. But I'm not confident enough that the logic is perfect every time to apply this across the board.